sbc_compassion/model/sponsorship_correspondance.py

```diff
@@ -0,0 +1,109 @@
+        related='sponsorship_id.correspondant_id', store=True)
+    child_id = fields.Many2one(related='sponsorship_id.child_id', store=True)
+    # Field used for identifying correspondence
+    kit_id = fields.Integer('Kit id', copy=False, readonly=True)
+    letter_type = fields.Selection(selection=[
+        ('S2B', _('Sponsor to beneficiary')),
+        ('B2S', _('Beneficiary to sponsor'))])
+    communication_type = fields.Selection(selection=[
+        ('scheduled', _('Scheduled')),
+        ('response', _('Response')),
+        ('thank_you', _('Thank you')),
+        ('third_party', _('Third party')),
+        ('christmas', _('Christmas')),
+        ('introduction', _('Introduction'))])
+    state = fields.Selection(selection=[
+        ('new', _('New')),
+        ('to_translate', _('To translate')),
+        ('translated', _('Translated')),
+        ('quality_checked', _('Quality checked')),
+        ('rejected', _('Rejected')),
+        ('sent', _('Sent')),
+        ('delivered', _('Delivered'))])
+    is_encourager = fields.Boolean()
+    mandatory_review = fields.Boolean(
+        related='sponsorship_id.correspondant_id.mandatory_review',
+        store=True)
+    letter_image = fields.Binary()
+    attachments_ids = fields.Many2many('ir.attachment')
+    physical_attachments = fields.Selection(selection=[
+        ('none', _('None')),
+        ('sent_by_mail', _('Sent by mail')),
+        ('not_sent', _('Not sent'))])
+    attachments_description = fields.Text()
+    supporter_languages_ids = fields.Many2many(
+        related='sponsorship_id.correspondant_id.spoken_langs_ids',
+        store=True)
+    beneficiary_language_ids = fields.Many2many(
+        related='sponsorship_id.child_id.project_id.country_id.\
+spoken_langs_ids', store=True)
+    # First spoken lang of partner
+    original_language_id = fields.Many2one(
+        'res.lang.compassion', compute='_set_current_language', store=True)
+    destination_language_id = fields.Many2one('res.lang.compassion')
+    template_id = fields.Selection(selection=[
+        ('template_1', _('Template 1')),
+        ('template_2', _('Template 2')),
+        ('template_3', _('Template 3')),
+        ('template_4', _('Template 4')),
+        ('template_5', _('Template 5'))])
+    original_text = fields.Text()
+    translated_text = fields.Text()
+    source = fields.Selection(selection=[
+        ('letter', _('Letter')),
+        ('email', _('E-Mail')),
+        ('website', _('Compassion website'))])
+
+    _sql_constraints = [
+        ('kit_id',
+         'unique(kit_id)',
+         _('The kit id already exists in database.'))
+    ]
+
+    ##########################################################################
+    #                             FIELDS METHODS                             #
+    ##########################################################################
+
+    @api.depends('sponsorship_id')
+    def _set_name(self):
+        self.name = str(
```

Comment:
Add a test if sponsorship_id is not defined, return `_('New correspondence')` (function `_()` is imported from openerp)